examples/magic_square.py

```diff
@@ -0,0 +1,477 @@
+                │             │               │
+(0, 2): ───H────┼@────────────@───H───@───────┼────────────
+                ││                    │       │
+(0, 3): ────────┼┼────────────────H───@───H───M────────────
+                ││
+(1, 0): ────────┼┼────────────────H───@───H───M('bob')─────
+                ││                    │       │
+(1, 1): ───H────@┼────H───H───@───H───@───────┼────────────
+                 │            │               │
+(1, 2): ───H─────@────H───────@───H───@───────┼────────────
+                                      │       │
+(1, 3): ──────────────────────────H───@───H───M────────────
+               └──┘
+
+
+Win rate for all 9 possible inputs, out of 10,000 repetitions each:
+[[1. 1. 1.]
+ [1. 1. 1.]
+ [1. 1. 1.]]
+
+"""
+
+
+import dataclasses
+
+import cirq
+import numpy as np
+
+
+def bell_pair_prep_circuit(q0: cirq.GridQubit, q1: cirq.GridQubit) -> cirq.Circuit:
+    """Prepare a Bell state between qubits q0 and q1.
+
+    Args:
+        q0: One qubit.
+        q1: The other qubit.
+
+    Returns:
+        A circuit creating a Bell state.
+    """
+    return cirq.Circuit(cirq.H.on_each(q0, q1), cirq.CZ(q0, q1), cirq.H(q1))
+
+
+def state_prep_circuit(
+    alice_data_qubits: tuple[cirq.GridQubit, cirq.GridQubit],
+    bob_data_qubits: tuple[cirq.GridQubit, cirq.GridQubit],
+) -> cirq.Circuit:
+    """Construct a circuit to produce the initial Bell pairs.
+
+    Args:
+        alice_data_qubits: Alice's data qubits.
+        bob_data_qubits: Bob's data qubits.
+
+    Returns:
+        A circuit preparing the Bell states.
+    """
+    a1, a2 = alice_data_qubits
+    b1, b2 = bob_data_qubits
+    return bell_pair_prep_circuit(a1, b1).zip(bell_pair_prep_circuit(a2, b2))
+
+
+def construct_measure_circuit(
+    alice_qubits: list[cirq.GridQubit],
+    bob_qubits: list[cirq.GridQubit],
+    mermin_row: int,
+    mermin_col: int,
+) -> cirq.Circuit:
+    """Construct a circuit to implement the measurement.
+
+    We use the conventions described in https://en.wikipedia.org/wiki/Quantum_pseudo-telepathy.
+    In particular, the Mermin-Peres table is
+
+     I ⊗ Z | Z ⊗ I | Z ⊗ Z
+     X ⊗ I | I ⊗ X | X ⊗ X
+    -X ⊗ Z |-Z ⊗ X | Y ⊗ Y
+
+    Args:
+        alice_qubits: The line of qubits to use for Alice (measure, data, data, measure).
+        bob_qubits: The line of qubits to use for Bob (measure, data, data, measure).
+        mermin_row: The row of the Mermin-Peres square to measure.
+        mermin_col: The column of the Mermin-Peres square to measure.
+
+    Returns:
+        A circuit implementing the measurement.
+    """
+
+    q = alice_qubits[1:3]  # data qubits
+    m = (alice_qubits[0], alice_qubits[3])  # measure qubits
+    if mermin_row == 0:
+        alice_circuit = cirq.Circuit(cirq.M(*q, key="alice"))
+    elif mermin_row == 1:
+        alice_circuit = cirq.Circuit(cirq.H.on_each(*q), cirq.M(*q, key="alice"))
+    elif mermin_row == 2:
+        alice_circuit = cirq.Circuit(
```

Comment:
you can do `Circuit.from_moments` and just list the operations without explictly creating moments e.g.
```py3
        alice_circuit = cirq.Circuit.from_moments(
            cirq.CZ(*q),
            cirq.H.on_each(*q, *m),
            cirq.CZ.on_each(*zip(m, q)),
            cirq.H.on_each(*m),
            cirq.M(*m, key="alice"),
    )
```